src/reactviews/pages/SchemaCompare/components/SchemaDifferences.tsx

```diff
@@ -3,35 +3,88 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { useContext } from "react";
+import * as React from "react";
+import { FixedSizeList as List, ListChildComponentProps } from "react-window";
 import {
-    Checkbox,
-    createTableColumn,
-    Table,
+    useScrollbarWidth,
+    useFluent,
     TableBody,
     TableCell,
-    TableColumnDefinition,
+    TableRow,
+    Table,
     TableHeader,
     TableHeaderCell,
-    TableRow,
+    createTableColumn,
+    useTableFeatures,
+    useTableSelection,
+    TableRowData as RowStateBase,
+    TableColumnDefinition,
+    Checkbox,
+    makeStyles,
+    Spinner,
 } from "@fluentui/react-components";
-import { FixedSizeList as List } from "react-window";
-import { schemaCompareContext } from "../SchemaCompareStateProvider";
 import { SchemaUpdateAction } from "../../../../sharedInterfaces/schemaCompare";
 import { locConstants as loc } from "../../../common/locConstants";
 import { DiffEntry } from "vscode-mssql";
+import { schemaCompareContext } from "../SchemaCompareStateProvider";
 
-type DiffItem = DiffEntry & {
-    id: number;
-};
+const useStyles = makeStyles({
+    HeaderCellPadding: {
+        padding: "0 8px",
+    },
+
+    PrimaryCheckboxPadding: {
+        paddingLeft: "16px",
+    },
+});
+
+interface TableRowData extends RowStateBase<DiffEntry> {
+    onClick: (e: React.MouseEvent) => void;
+    onKeyDown: (e: React.KeyboardEvent) => void;
+    selected: boolean;
+    appearance: "brand" | "none";
+}
+
+interface ReactWindowRenderFnProps extends ListChildComponentProps {
+    data: TableRowData[];
+}
 
 interface Props {
     onDiffSelected: (id: number) => void;
 }
 
-const SchemaDifferences = ({ onDiffSelected }: Props) => {
-    const context = useContext(schemaCompareContext);
+export const SchemaDifferences = ({ onDiffSelected }: Props) => {
+    const classes = useStyles();
+    const { targetDocument } = useFluent();
+    const scrollbarWidth = useScrollbarWidth({ targetDocument });
+    const context = React.useContext(schemaCompareContext);
     const compareResult = context.state.schemaCompareResult;
+    let [allDiffsIncluded, setAllDiffsIncluded] = React.useState(true);
+    let [someDiffsExcluded, setSomeDiffsExcluded] = React.useState(false);
+    let [allDiffsExcluded, setAllDiffsExcluded] = React.useState(false);
```

Comment:
This strikes me as a good opportunity for string union instead: `"allIncluded" | "allExcluded" | "someExcluded"`.  Three independent booleans seems ripe for bugs when they get out-of-sync.